Sofa/Component/Mapping/NonLinear/src/sofa/component/mapping/nonlinear/RigidRigidMapping.inl

```diff
@@ -132,26 +132,35 @@ void RigidRigidMapping<TIn, TOut>::init()
 
         if(globalToLocalCoords.getValue() == true)
         {
-            const typename In::VecCoord& xfrom =this->fromModel->read(core::ConstVecCoordId::position())->getValue();
+            const typename In::VecCoord& xfrom = this->fromModel->read(core::ConstVecCoordId::position())->getValue();
             switch (repartition.getValue().size())
             {
             case 0 :
                 for (i = 0; i < x.size(); i++)
                 {
-                    // pts[i] = x[i] - xfrom[0];
                     pts[i].getCenter() = xfrom[index.getValue()].getOrientation().inverse().rotate( x[i].getCenter() - xfrom[index.getValue()].getCenter() ) ;
                     pts[i].getOrientation() = xfrom[index.getValue()].getOrientation().inverse() * x[i].getOrientation() ;
                 }
                 break;
             case 1 :
                 for (i=0; i<xfrom.size(); i++)
+                {
                     for(sofa::Index j=0; j<repartition.getValue()[0]; j++,cpt++)
-                        pts[cpt] = x[cpt] - xfrom[i];
+                    {
```

Comment:
To be honest I don't like this `repartition` option, I prefer `rigidIndexPerPoint` that we have in the `RigidMapping` component. I'll be happy to add this option if you agree (although I don't like the idea of having all those options available).

How does `repartition` work. If `repartition=[2]`, each point of the parent will be mapped with two points of the child. If `repartition=[0, 2, 1, 3]` each point of the parent will be mapped with the corresponding number of points of the child (i.e. parent[i] will be mapped with repartition[i] points of the child). The main problem (apart from being difficult to understand) is that the mapping imposes that your child's points are ordered w.r.t the mapping. 

In comparison, with `rigidIndexPerPoint`, you specify the parent's point (index) for each point of the child. So here, it's possible to map the last point of the child with the first point of the parent.

So to answer your question... yes we need two indices here : `j` to iterate over the number of child's points to map to the `parent[i]`, and `cpt` to get the corresponding child point.

Reply:
I would even suggest to merge `RigidMapping` and `RigidRigidMapping`...

Reply:
What do you think about adding the template `Rigid3,Rigid3` to RigidMapping and seting RigidRigidMapping as deprecated?

Btw, I've been looking at the code of RigidMapping, and I see that `repartition` has been set as deprecated for this component. So merging both instead of updating RigidRigidMapping really makes more sense to me.

Reply:
Is this discussion resolved ?

Reply:
Well yes, I've explained why we need two indices `j` and `cpt`.